### ISSUE TITLE
Add multi-machine and Enterprise scenario support for System.Net tests

### ISSUE THREAD

**Issue by CIPop** [dotnet/corefx#9048]
/cc @himadrisarkar @davidsh @vijaykota @shrutigarg @markwilkie @MattGal 

System.Net.* testing requires multiple machines to exercise real-world networking scenarios. 
These include both the outer-loop tests as well as tests requiring more complicated setup and fixed SKUs such as perf.

Here is a list of requirements that our team brainstormed a while back:

## Network Scenarios
* Internet scenarios
  * Use a set of publicly available, always available Azure servers. 
  * Allow `localhost`, local network or private Azure servers.

* Enterprise scenarios (auth, IPSec)
  * Use `localhost` servers
  * Use local network servers
  * Use an Azure closed test environment

* Performance/stress scenarios

## Test Server Requirements
  
* Scripts should allow deployment on `localhost` as well as a specific on-prem machine or an Azure account.
An example of such deployment is available at https://github.com/davidsh/corefx-net-testservers.

* Server configuration scripts (both for local and remote deployments)
  * Use Bash/Powershell
  * Scripts should be able to both install and uninstall the server.
  * Ensure a clear state of the machine after uninstall.

## Test Client Requirements

* Standardized script naming and location will allow for automatic prerequisite installation/cleanup.
* Client prerequisite scripts: install and cleanup
  * Use Powershell/Bash
  * Will run as `Administrator` / `root`.

## Test Execution

* Endpoint configuration will be made through Environment variables (see #2383).
* A single script will be used to:
  * set the environment variables (a `.gitignore`d override file could allow developers to set up their own custom environment)
  * detect and install prerequisites
  * run selected tests
  * uninstall prerequisites
* Proposed prerequisite script location and naming:
  * contract_name\tests\scripts\client
    * setupclient.ps1(.sh) [-u(ninstall)]

  * contract_name\tests\scripts\server
    * setupserver.ps1(.sh) [-u(ninstall)]

**Comment by clrjunkie** [comment]
Regarding setupclient.ps/cleanupclient.ps1
Consider using one script that has /u uninstall switch. Having install and uninstall logic in one file is simpler to maintain and reduces the chance of forgetting to properly update the inverse logic.

**Comment by CIPop** [comment]
> Regarding setupclient.ps/cleanupclient.ps1

@clrjunkie Good point: updated above. (For the bash version it'll probably be --uninstall or -u while in Powershell it might be just -uninstall.)

**Comment by CIPop** [comment]
@ericeil @stephentoub PTAL

Through the PRs I've tried to check-in scripts that would allow anyone to deploy a 3 box environment: 
* Active Directory Domain Controller
* IIS Server (HTTP / WS)
* Client machine

All of these can be deployed by using the free Windows Server 2016 TP5 ISO in a VM host of choice.

At the end of deployment all the machines are part of the same domain, a domain user has been created and the appropriate `Configuration.*.cs` environment variables have been added to the Client role to enable capabilities that are now disabled in the CI runs (due to Azure public setup limitations).

Next steps would be to figure out how to integrate this with the CI system (I'm thinking for Outerloop only first) and create/auto-generate a `setup_client.sh` script to allow *NIX boxes to be part of the same environment.

We should also think of adding a Web-Proxy server and figure out how to run and collect differential information from e2e performance simulation tests.
